java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -100,6 +99,7 @@ public MaximumStepsReachedException(String s) {
 
   public ExplodedGraphWalker(JavaFileScannerContext context) throws MaximumStepsReachedException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of thrown exception 'org.sonar.java.se.ExplodedGraphWalker$MaximumStepsReachedException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)